examples/dodo_prompt.rs

```diff
@@ -77,8 +82,73 @@ impl Completer for CommentCompleter {
     }
 }
 
+pub struct NewlineForBackslashAndOpenDelimRule;
+
+impl NewlineRule for NewlineForBackslashAndOpenDelimRule {
+    fn evaluate_on_newline(&self, buf: &Buffer) -> bool {
+        last_non_ws_char_was_not_backslash(buf) && check_balanced_delimiters(buf)
+    }
+}
+
+pub fn check_balanced_delimiters(buf: &Buffer) -> bool {
+    let buf_vec = buf.range_graphemes_all();
+    let mut stack = vec![];
+    let mut symmetric_delimiters = HashSet::new();
```

Comment:
As above, this also looks like it could just be a counter.

Reply:
see other response.